integration-cli/docker_cli_daemon_test.go

```diff
@@ -496,6 +498,176 @@ func (s *DockerSuite) TestDaemonExitOnFailure(c *check.C) {
+		d.Stop()
+		c.Fatal("--bridge option with an external bridge that is not present should cause the daemon to error out with a failure")
+	}
+	brctlCmd := exec.Command("brctl", "addbr", "external-bridge")
+	_, _, _, err := runCommandWithStdoutStderr(brctlCmd)
+	if err != nil {
+		c.Fatal("failed to create a bridge")
+	}
+
+	bridgeIp := "192.169.1.1/24"
+	_, bridgeIPNet, _ := net.ParseCIDR(bridgeIp)
+
+	brctlCmd = exec.Command("ifconfig", "external-bridge", bridgeIp, "up")
+	_, _, _, err = runCommandWithStdoutStderr(brctlCmd)
+	if err != nil {
+		c.Fatal("failed to assign ip-address to the bridge")
+	}
+
+	if err = d.StartWithBusybox("--bridge", "external-bridge"); err != nil {
+		c.Fatalf("--bridge option with a valid External-bridge should not fail : %s", err.Error())
+	}
+
+	ipTablesSearchString := bridgeIPNet.String()
+	ipTablesCmd := exec.Command("iptables", "-t", "nat", "-nvL")
+	out, _, err := runCommandWithOutput(ipTablesCmd)
+	if err != nil {
+		c.Fatalf("Could not run iptables -nvL: %s, %v", out, err)
+	}
+
+	if !strings.Contains(out, ipTablesSearchString) {
+		c.Fatalf("iptables output should have contained %q, but was %q", ipTablesSearchString, out)
+	}
+
+	out, err = d.Cmd("run", "-d", "--name", "ExtContainer", "busybox", "top")
+	if err != nil {
+		c.Fatalf("Could not run a Container : %s", err.Error())
+	}
+
+	containerIp := d.findContainerIP(c, "ExtContainer")
+	ip := net.ParseIP(containerIp)
+	if !bridgeIPNet.Contains(ip) {
+		c.Fatalf("Container IP-Address must be in the same subnet range as the external bridge : %s", containerIp)
+	}
+
+	// Reset to Defaults
+	deleteBridge(c, "external-bridge")
+	d.Restart()
+}
+
+func deleteBridge(c *check.C, bridge string) {
+	ifCmd := exec.Command("ifconfig", bridge, "down")
+	_, _, _, err := runCommandWithStdoutStderr(ifCmd)
+	if err != nil {
+		c.Fatalf("Could not bring down %s bridge interface %s", bridge, err.Error())
+	}
+
+	ifCmd = exec.Command("brctl", "delbr", bridge)
+	_, _, _, err = runCommandWithStdoutStderr(ifCmd)
+	if err != nil {
+		c.Fatalf("Could not delete docker0 Bridge %s", err.Error())
+	}
+	flushCmd := exec.Command("iptables", "-t", "nat", "--flush")
+	_, _, _, err = runCommandWithStdoutStderr(flushCmd)
+	if err != nil {
+		c.Fatalf("Could not flush iptables %s", err.Error())
+	}
+}
+
+func (s *DockerSuite) TestDaemonBridgeIP(c *check.C) {
+	// TestDaemonBridgeIP Steps
+	// 1. Delete the existing docker0 Bridge
+	// 2. Set --bip daemon configuration and start the new Docker Daemon
+	// 3. Check if the bip config has taken effect using ifconfig and iptables commands
+	// 4. Launch a Container and make sure the IP-Address is in the expected subnet
+	// 5. Delete the docker0 Bridge
+	// 6. Restart the Docker Daemon (with no --bip settings)
+	//    This Restart takes care of bringing docker0 interface back to auto-assigned IP
+	// 7. Stop the Docker Daemon (via defered action)
+
+	defaultNetworkBridge := "docker0"
```

Comment:
Yes. it is brutal. There isn't an easy answer to this. 
But, the above steps actually performs everything required for testing bip and resets the 
environment back to the innocent looking docker0 and also I added a ping test to confirm everything is back to normal.